docs/advanced.rst

```diff
@@ -1589,24 +1589,56 @@ It is also possible to call python functions via ``operator()``.
     py::object result_py = f(1234, "hello", some_instance);
```

Comment:
Yea, agreed, most of this library now falls under "Advanced", and things like objects in pytypes.h are scarcely mentioned.

Splitting into sections could be good -- eg wrapper types (dict, tuple etc), handle/object, utility (print), eigen, NumPy (I'd be willing to rewrite the docs on this one).